offsetPrint.asm

```diff
@@ -1,60 +1,193 @@
+StringEnd: var #1
+static StringEnd + #0, #'\0'
 
+LaneSize: var #1
+static LaneSize + #0, #5
+
+FirstLaneBasePos: var #5
+static FirstLaneBasePos + #0, #40
+static FirstLaneBasePos + #1, #80
+static FirstLaneBasePos + #2, #120
+static FirstLaneBasePos + #3, #160
+static FirstLaneBasePos + #4, #200
+
+; >>>>>>>>>>>> STRINGS
+TestLaneChars: var #5
+TestLaneChar0: string ">a=f=k=p=FIRST LINE OF TEST LANE=t=y=d=<"
+TestLaneChar1: string ">b=g=l=?=SCOND LINE OF TEST LANE=u=z=e=<"
+TestLaneChar2: string ">c=h=m=q=THIRD LINE OF TEST LANE=v=a=f=<"
+TestLaneChar3: string ">d=i=n=r=FORTH LINE OF TEST LANE=w=b=g=<"
+TestLaneChar4: string ">e=j=o=s=FIFTH LINE OF TEST LANE=x=c=h=<"
+
+; >>>>>>>>>>>> GLOBAL VARIABLES
+globalOffset: var #2          ; current line offset = [positive/left, negative/right]
+currentLaneBase: var #1       ; current lane base positions
+
+; >>>>>>>>>>>> FUNCTIONS
+
+; === MAIN ===
+main:
+  call globalInit             ; init global variables
+
+  loadn r0, #FirstLaneBasePos ; r0 = FirstLaneBasePos[]
+  store currentLaneBase, r0   ; currentLaneBase = FirstLaneBasePos[]
+
+mainLoop:
   breakp
-  jmp mainPrint
+  call printLane              ; call print first lane
+  call updateOffset           ; call update offset
+  jmp mainLoop                ; test loop
+
+  halt                        ; end of program
+
+
+; === GLOBALINIT ===
+globalInit:                   ; === GLOBALINIT ===
+  push r0                     ; protecting registers
+  push r1
+
+  loadn r0, #globalOffset     ; r0 = *globalOffset[0]
+  loadn r1, #10                ; default positive global offset
+  storei r0, r1               ; globalOffset[0] = positive offset
+
+  inc r0                      ; r0 = *globalOffset[1]
+  loadn r1, #30               ; default negative global offset
+  storei r0, r1               ; globalOffset[1] = negative offset
+
+globalInitLanes:              ; === GLOBALINIT > LANES ===
+  ; TestLane
+  loadn r0, #TestLaneChars    ; r0 = TestLaneChars[0]
+  loadn r1, #TestLaneChar0    ; r1 = *TestLaneChar0
+  storei r0, r1               ; TestLaneChars[0] = TestLaneChar0
+
+  inc r0                      ; r0 = TestLaneChars[1]
+  loadn r1, #TestLaneChar1    ; r1 = *TestLaneChar1
+  storei r0, r1               ; TestLaneChars[1] = TestLaneChar1
+
+  inc r0                      ; r0 = TestLaneChars[2]
+  loadn r1, #TestLaneChar2    ; r1 = *TestLaneChar2
+  storei r0, r1               ; TestLaneChars[2] = TestLaneChar2
+
+  inc r0                      ; r0 = TestLaneChars[3]
+  loadn r1, #TestLaneChar3    ; r1 = *TestLaneChar3
+  storei r0, r1               ; TestLaneChars[3] = TestLaneChar3
+
+  inc r0                      ; r0 = TestLaneChars[4]
+  loadn r1, #TestLaneChar4    ; r1 = *TestLaneChar4
+  storei r0, r1               ; TestLaneChars[4] = TestLaneChar4
+
+globalInitEnd:                ; === GLOBALINIT > END ===
+  pop r1                      ; recovering registers
+  pop r0
+
+  rts                         ; return globalInit function
```

Comment:
Acho que pode melhorar o comentário dessa linha, talvez "return of globalInit function"